crates/nargo_cli/src/cli/codegen_verifier_cmd.rs

```diff
@@ -19,8 +21,22 @@ pub(crate) fn run(args: CodegenVerifierCommand, config: NargoConfig) -> Result<(
     #[allow(deprecated)]
     let smart_contract_string = backend.eth_contract_from_cs(compiled_program.circuit);
 
-    let contract_dir = config.program_dir.join(CONTRACT_DIR);
-    create_named_dir(&contract_dir, "contract");
+    let manifest_path = config.program_dir.join(PKG_FILE);
+    if !manifest_path.exists() {
+        return Err(CliError::MissingTomlFile("Nargo".to_owned(), manifest_path));
+    }
+
+    let manifest = match manifest::parse(manifest_path) {
+        Ok(it) => it,
+        Err(err) => return Err(CliError::Generic(err.to_string())),
```

Comment:
it may not be the best for this situation, but I couldn't find other way to parse(or access) the manifest file...🤔 

Could you guide me on a recommended way to convert `InvalidPackageError` to `CliError`?

Reply:
This is the correct function to use but rather than converting the `InvalidPackageError` to a generic error we should add a new error type to `CliError` for the situation where someone is trying to interact with a Nargo project with a bad `Nargo.toml` file.

Let's add an entry which looks something like this:
```rust
/// Error while loading package manifest.
#[error(transparent)]
InvalidManifest(#[from] InvalidPackageError),
```

We can then replace the `match` statement with `let manifest = manifest::parse(manifest_path)?;`